fe/fe-core/src/main/java/com/starrocks/connector/trino/TrinoViewColumnTypeConverter.java

```diff
@@ -0,0 +1,193 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.trino;
+
+import com.starrocks.catalog.ArrayType;
+import com.starrocks.catalog.MapType;
+import com.starrocks.catalog.PrimitiveType;
+import com.starrocks.catalog.ScalarType;
+import com.starrocks.catalog.StructField;
+import com.starrocks.catalog.StructType;
+import com.starrocks.catalog.Type;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.starrocks.connector.ColumnTypeConverter.getCharLength;
+import static com.starrocks.connector.ColumnTypeConverter.getPrecisionAndScale;
+import static com.starrocks.connector.ColumnTypeConverter.getTypeKeyword;
+import static com.starrocks.connector.ColumnTypeConverter.getVarcharLength;
+import static com.starrocks.connector.ColumnTypeConverter.splitByFirstLevel;
+
+public class TrinoViewColumnTypeConverter {
+
+    public static final String COMPLEX_PATTERN = "([0-9a-z<>(),:_ \"]+)";
+    public static final String ARRAY_PATTERN = "^array\\(" + COMPLEX_PATTERN + "\\)";
+
+    public static final String MAP_PATTERN = "^map\\(" + COMPLEX_PATTERN + "\\)";
+
+    public static final String STRUCT_PATTERN = "^row\\(" + COMPLEX_PATTERN + "\\)";
+
+    private static final List<String> UNSUPPORTED_TYPES = Arrays.asList("UNIONTYPE");
+
+    public static Type fromTrinoType(String trinoType) {
```

Comment:
How is this different from ColumnTypeConverter.fromHiveType()

Reply:
The float, array,struct,map are different from ColumnTypeConverter.fromHiveType()